### PR TITLE
Add default settings

### PR DESCRIPTION
**Pull request by terokarvinen**
Add default settings
    
Default settings allow using LSP plugin without modifying settings.json. Most users are not likely to figure out and test support for new language servers, so they are likely to use default settings.
    
This makes plugin installation process very simple. Installing micro with LSP support for Go becomes a two step process. E.g. for Debian: 'sudo apt-get install micro golang-go gopls; micro --plugin install lsp'.
    
Some versions of settings.json examples on README.md don't mention all tested language servers. I added all language servers mentioned anywhere in the README, including Rust rls and Lua lua-lsp. I did not test that these language servers work, I simply added their names and filetypes from README.

### CODE REVIEW

File: main.lua [hunk 1]
```diff
@@ -55,8 +55,11 @@ end
 function startServers()
 	local wd, _ = os.Getwd()
 	rootUri = fmt.Sprintf("file://%s", wd)
-	local envFallback, _ = os.Getenv("MICRO_LSP")
-	local server = mysplit(config.GetGlobalOption("lsp.server") or envFallback or '', ",")
+	local fallback, _ = os.Getenv("MICRO_LSP")
+	if ("" == fallback) then
+		fallback = 'python=pylsp,go=gopls,typescript=deno,rust=rls,lua=lua-lsp lsp={"importMap": "./import_map.json"}'
```
Comment (AndCake) [inline main.lua:60]: The part with `={"importMap":"./import_map.json"}` is just an example of how to transfer additional initialization options to an LSP. For a fallback, I would leave it out. 

The above line should be the following instead:

```
  fallback = 'python=pylsp,go=gopls,typescript=deno lsp,javascript=deno lsp,rust=rls,lua=lua-lsp'
```

I have to try out to see if any error messages would be shown if an LSP is not actually there.

File: main.lua [hunk 2]
```diff
@@ -1,4 +1,4 @@
-VERSION = "0.4.3"
+VERSION = "0.4.4"
```
Comment (AndCake) [inline main.lua:1]: I would undo this, since I wanted to release this together with the references feature, making it a 0.5.0

**Review by AndCake — COMMENTED** [review]
Apart from the small suggestions I added, this looks good!

File: main.lua (hunk 1 above)
Comment (AndCake) [reply main.lua:60]: Luckily if a language server does not exist, it seems it just gracefully says "onExit <nil>" in the log.txt and no error message is shown in the UI, so this is good! Seems like only the fallback itself needs to be adapted :)

**Comment by terokarvinen** [comment]
Good point @AndCake, the import_map should be left out.  And Javascript support seems to come with deno, good addition. 

```lua
fallback = 'python=pylsp,go=gopls,typescript=deno lsp,javascript=deno lsp,rust=rls,lua=lua-lsp'
```

I wanted to confirm that the value of both "typescript" and "javascript" is "deno lsp", and the space does not need to be escaped, is this correct?

OK, so 0.4.3 it is.

**Comment by AndCake** [comment]
Yes so basically deno is a "native" typescript interpreter. And since typescript is a superset of JavaScript, it can actually do both. The space between `deno` and `lsp` does indeed not need to be escaped. It is treated as a parameter to the program.

**Comment by terokarvinen** [comment]
Thanks for the advice, I made the changes.

**Review by AndCake — APPROVED** [review]

**Pull request merged**